docs/prio-processor-guide.md

```diff
@@ -0,0 +1,371 @@
+| `SHARED_SECRET`                  | A shared secret generated by `prio shared-seed`.                                 |
+| `PRIVATE_KEY_HEX`                | The private key of the processor as a hex binary string.                         |
+| `PUBLIC_KEY_HEX_INTERNAL`        | The public key of the processor as a hex binary string.                          |
+| `PUBLIC_KEY_HEX_EXTERNAL`        | The public key of the co-processor as a hex binary string.                       |
+| `BUCKET_INTERNAL_PRIVATE`        | The bucket containing data that is viewable by the processor alone.              |
+| `BUCKET_INTERNAL_SHARED`         | The bucket containing data from the processor's previous stage.                  |
+| `BUCKET_EXTERNAL_SHARED`         | The bucket containing incoming data from the co-processor's previous stage.      |
+| `GOOGLE_APPLICATION_CREDENTIALS` | The path on the container filesystem containing GCP service account credentials. |
+
+Data configuration should be mounted into the `/app/processor/config` directory
+and set via `DATA_CONFIG`. Likewise, the GCP service account JSON key-file
+should be mounted into `/app/.credentials` and set via
+`GOOGLE_APPLICATION_CREDENTIALS`.
+
+### Building an image from source
+
+To build the container locally:
+
+```bash
+make build
+```
+
+This will generate two images that are ready to use. The development image is
+configured to run unit and integration tests, while the production image will
+initialize the single trigger, batched-processing mode.
+
+```bash
+# run the tests
+docker run prio:dev
+
+# start a shell session
+# --interactive --tty
+docker run -it prio:dev bash
+
+# start the server
+docker run prio:latest
+```
+
+See the prio-processor README for more details about the development
+environment.
+
+### python-libprio
+
+`python-libprio` is a Python3 extension module that wraps the Prio routines. It
+implements a SWIG interface for building Python wrappers.
+
+The wrapper generated by the SWIG interface file mirrors the structure of code
+in "mprio.h".
+
+```python
+from prio.libprio import *
+
+data_in = ...   # <class 'bytes'>
+server = ...    # <class 'PyCapsule'>
+
+verifier = PrioVerifier_new(server)
+packet = PrioPacketVerify1_new()
+
+PrioVerifier_set_data(verifier, data)
+PrioPacketVerify1_set_data(packet, verifier)
+
+data_out = PrioPacketVerify1_write(packet)
+```
+
+The `prio` module provides an higher-level API that handles initialization,
+reference-counting and serialization.
+
+```python
+from prio.prio import *
+import pickle
+
+data_in = ...   # <class 'bytes'>
+server = ...    # <class 'Server'>
+
+with Prio() as p:
+    verifier = p.Verifier(server, data_in)
+    data_out = pickle.dumps(verifier.create_verify1())
+```
+
+These libraries are used to build up a series of examples that include
+asynchronous I/O, multi-processing, and container orchestration.
+
+### The `prio` cli tool
```

Comment:
I think I would incorporate any content here into the command-line reference, and just talk about prio-processor here. The prio command line, again, seems like an part of the implementation of the higher-level processor application. Might be useful to *reference* that document here (in case things go wrong), but I don't think we need to duplicate the content here.